MVVM Module/Tests/Feed UI/FeedUIIntegrationTests.swift

```diff
@@ -47,6 +47,31 @@ final class FeedUIIntegrationTests: XCTestCase {
 		XCTAssertFalse(sut.isShowingLoadingIndicator, "Expected no loading indicator once user initiated loading completes with error")
 	}
 	
+	func test_loadFeed_BadRequestError() {
+		let (sut, loader) = makeSUT()
+		
+		sut.loadViewIfNeeded()
+		loader.completeFeedLoading(at: 0)
+		sut.simulateUserInitiatedFeedReload()
+		
+		let errors = [403: "BAD_REQUEST_ERROR",
+					  401: "UNAUTHORIZED_ERROR",
+					  404: "SERVER_NOT_FOUND_ERROR",
+					  503: "SERVICE_UNAVAILABLE_ERROR",
+					  504: "TIMEOUT_ERROR",
+					  505: "OTHER_ERROR"]
+		
+		errors.forEach { (key, value) in
+			let customErrorMessage = localized(value)
+			
+			let serverError = NSError(domain: "Some error", code: key, userInfo: nil)
+			loader.completeFeedLoadingWithError(error: serverError, at: 1)
+			
+			XCTAssertEqual(sut.errorView?.message, customErrorMessage, "Expected errorLabel's text \(customErrorMessage), received \(String(describing: sut.errorView?.message))")
+			XCTAssertEqual(sut.errorView?.isHidden, false, "Expected errorView to be visible")
```

Comment:
You can create a `sut?.errorMessage` test helper method to decouple the test from the view hierarchy. So you can change the view hierarchy later without breaking the tests.

Moreover, `XCTAssertEqual` already logs the expected value, so the assertion messages could be shorter.

For example:

```suggestion
			XCTAssertEqual(sut?.errorMessage, customErrorMessage)
			XCTAssertEqual(sut?.errorView?.isHidden, false, "Expected errorView to be visible")
```